src/flb_log.c

```diff
@@ -454,6 +478,78 @@ int flb_log_set_file(struct flb_config *config, char *out)
     return 0;
 }
 
+
+/*
+ * Create and register cmetrics for the runtime logger.
+ * The caller must free the returned struct using flb_log_metrics_destroy.
+ */
+struct flb_log_metrics *flb_log_metrics_create()
+{
+    struct flb_log_metrics *metrics;
+    int i;
+    const char *message_type_str;
+    uint64_t ts;
+    int ret;
+
+    metrics = flb_calloc(1, sizeof(struct flb_log_metrics));
+    if (metrics == NULL) {
+        flb_errno();
+        return NULL;
+    }
+
+    metrics->cmt = cmt_create();
+    if (!metrics->cmt) {
+        flb_free(metrics);
+        return NULL;
+    }
+
+    metrics->logs_total_counter = cmt_counter_create(metrics->cmt,
+                                                     "fluentbit",
+                                                     "logger",
+                                                     "logs_total",
+                                                     "Total number of logs",
+                                                     1, (char *[]) {"message_type"});
+    if (metrics->logs_total_counter == NULL) {
+        cmt_destroy(metrics->cmt);
+        flb_free(metrics);
+        return NULL;
+    }
+
+    /*
+     * Initialize counters for all log message types to 0.
+     * This assumes types are contiguous starting at 1 (FLB_LOG_ERROR).
+     */
+    ts = cfl_time_now();
+    for (i = 1; ; i++) {
+        message_type_str = flb_log_message_type_str(i);
+        if (!message_type_str) {
+            break;
+        }
+
+        ret = cmt_counter_set(metrics->logs_total_counter,
+                              ts,
+                              0,
+                              1, (char *[]) {message_type_str});
+        if (ret == -1) {
+            cmt_counter_destroy(metrics->logs_total_counter);
```

Comment:
Please replace this cleanup block (and the one before) with a call to `flb_log_metrics_destroy`

Reply:
I've been getting a lot of mixed messages around cleanup patterns. Not only are there different patterns across the code, I've had different people telling me to use different conflicting patterns.

It would be good to have memory management patterns written down in CONTRIBUTING.md because the PR review churn is rough.

For now, I've updated the free function to be conditional:
```c
/* Frees the metrics instance and its associated resources. */
void flb_log_metrics_destroy(struct flb_log_metrics *metrics)
{
    if (metrics != NULL && metrics->cmt != NULL) {
        cmt_destroy(metrics->cmt);
    }
    if (metrics != NULL) {
        flb_free(metrics);
    }
}
```